### PR TITLE
add migration and Db models

### PR DESCRIPTION
**Pull request by borovHokage**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/862

### CODE REVIEW

File: src/HistoryService.Data.Provider.MsSql.Ef/HistoryServiceDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿using LT.DigitalOffice.HistoryService.Models.Db;
+using Microsoft.EntityFrameworkCore;
+using System.Reflection;
+
+namespace LT.DigitalOffice.HistoryService.Data.Provider.MsSql.Ef
+{
+    public class HistoryServiceDbContext : DbContext, IDataProvider
+    {
+        public DbSet<DbService> Services { get; set; }
+        public DbSet<DbServicesHistories> ServiceHistories { get; set; }
```
Comment (MariyaSin) [inline src/HistoryService.Data.Provider.MsSql.Ef/HistoryServiceDbContext.cs]: DbServiceHistory, variable ServicsesHistories

File: src/HistoryService.Data.Provider.MsSql.Ef/Migrations/20210825160103_InitialTables.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    table.PrimaryKey("PK_Services", x => x.Id);
+                });
+
+            builder.AddUniqueConstraint(
+                name: $"UX_Name_unique",
+                table: DbService.TableName,
+                column: nameof(DbService.Name));
+
+            builder.CreateTable(
+                name: DbServicesHistories.TableName,
```
Comment (MariyaSin) [inline src/HistoryService.Data.Provider.MsSql.Ef/Migrations/20210825160103_InitialTables.cs]: DbServiceHistory

File: src/HistoryService.Data.Provider/IDataProvider.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+using LT.DigitalOffice.Kernel.Enums;
+using Microsoft.EntityFrameworkCore;
+
+namespace LT.DigitalOffice.HistoryService.Data.Provider
+{
+    [AutoInject(InjectType.Scoped)]
+    public interface IDataProvider : IBaseDataProvider
+    {
+        DbSet<DbService> Services { get; set; }
+        DbSet<DbServicesHistories> ServiceHistories { get; set; }
```
Comment (MariyaSin) [inline src/HistoryService.Data.Provider/IDataProvider.cs]: DbServiceHistory, variable ServicesHistories

File: src/HistoryService.Models.Db/DbService.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        public Guid Id { get; set; }
+        public string Name { get; set; }
+        public Guid CreatedBy { get; set; }
+        public DateTime CreatedAtUtc { get; set; }
+        public Guid? ModifiedBy { get; set; }
+        public DateTime? ModifiedAtUtc { get; set; }
+
+        [IgnoreParse]
+        public ICollection<DbServicesHistories> ServiceHistories { get; set; }
```
Comment (MariyaSin) [inline src/HistoryService.Models.Db/DbService.cs]: DbServiseHistory, variable ServicesHistories

File: src/HistoryService.Models.Db/DbService.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+            builder.
+                HasKey(p => p.Id);
+
+            builder
+                .Property(p => p.Name)
+                .IsRequired();
+
+            builder
+                .HasMany(s => s.ServiceHistories)
```
Comment (MariyaSin) [inline src/HistoryService.Models.Db/DbService.cs]: ServicesHistories

File: src/HistoryService.Models.Db/DbServicesHistories.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+
+namespace LT.DigitalOffice.HistoryService.Models.Db
+{
+    public class DbServicesHistories
+    {
+        public const string TableName = "ServiceHistories";
```
Comment (MariyaSin) [inline src/HistoryService.Models.Db/DbServicesHistories.cs]: ServicesHistories

File: src/HistoryService.Data.Provider.MsSql.Ef/HistoryServiceDbContext.cs (hunk 1 above)
Comment (borovHokage) [reply src/HistoryService.Data.Provider.MsSql.Ef/HistoryServiceDbContext.cs]: done

File: src/HistoryService.Data.Provider/IDataProvider.cs (hunk 3 above)
Comment (borovHokage) [reply src/HistoryService.Data.Provider/IDataProvider.cs]: done

File: src/HistoryService.Data.Provider.MsSql.Ef/Migrations/20210825160103_InitialTables.cs (hunk 2 above)
Comment (borovHokage) [reply src/HistoryService.Data.Provider.MsSql.Ef/Migrations/20210825160103_InitialTables.cs]: done

File: src/HistoryService.Models.Db/DbService.cs (hunk 4 above)
Comment (borovHokage) [reply src/HistoryService.Models.Db/DbService.cs]: done

File: src/HistoryService.Models.Db/DbService.cs (hunk 5 above)
Comment (borovHokage) [reply src/HistoryService.Models.Db/DbService.cs]: done

File: src/HistoryService.Models.Db/DbServicesHistories.cs (hunk 6 above)
Comment (borovHokage) [reply src/HistoryService.Models.Db/DbServicesHistories.cs]: done

File: Common.Configurations.targets [hunk 7]
```diff
@@ -33,7 +33,8 @@
   </ItemGroup>
 
   <ItemGroup Condition="$(AssemblyName.Contains('.UnitTests')) != 'true'">
-    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.21" />
+    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.1.35" />
+	<PackageReference Include="LT.DigitalOffice.Models" Version="1.0.31" />
```
Comment (BelehovEgor) [inline Common.Configurations.targets]: remove extra tab

Comment (BelehovEgor) [reply Common.Configurations.targets]: and please use kernel 37, version 35 is not correct

**Review by BelehovEgor — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Review by Cheephot — APPROVED** [review]

File: Common.Configurations.targets (hunk 7 above)
Comment (borovHokage) [reply Common.Configurations.targets]: done

**Pull request merged**